x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/common/SparseFileTracker.java

```diff
@@ -464,8 +474,20 @@ private void onGapSuccess(final Range gapRange) {
     private void maybeUpdateCompletePointer(Range gapRange) {
         assert Thread.holdsLock(ranges);
         if (gapRange.start == 0) {
```

Comment:
Should we also change this (can be a separate PR if you prefer) to trigger when `gapRange.start <= complete && gapRange.end > complete`? Seems it does not support fetch from indexing tier well today.

Reply:
There is possibly more work to do, like supporting out of order range fetching and still updating the complete pointer correctly. That is definitely a separate PR.

Reply:
Disregard this, the merging of ranges done in `onGapSuccess` means this works as is.